eng/pipelines/common/global-build-job.yml

```diff
@@ -70,6 +71,13 @@ jobs:
 
       - name: _osParameter
         value: -os ${{ parameters.osGroup }}
+      - name: _archParameter
+        value: -arch ${{ parameters.archType }}
+      - ${{ if and(eq(parameters.runtimeFlavor, 'coreclr'), in(parameters.osGroup, 'ios', 'iossimulator', 'tvos', 'tvossimulator', 'maccatalyst')) }}:
```

Comment:
@MichalStrehovsky how risky would it be to introduce nativeaot as a `RuntimeFlavor` variant? 
It seems to me like it would simplify (and avoid confusion) in few places.

Reply:
We've been avoiding calling it a flavor because most of the time we want the CoreCLR codepaths. LLVM-fullAOT is not a flavor either. There's a `runtimeVariant` that we currently don't use. I don't know how much better/worse it would be if we use that. We haven't had much need to pass this around in yaml and in the places we do, we just pass it as a bool (`nativeAotTest`) because readytorun already used bools (`readyToRun` and `compositeBuildMode`).